src/compas_fab/datastructures/plan.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from collections import OrderedDict
+from itertools import count
+
+from compas.utilities import DataEncoder
+from compas.utilities import DataDecoder
+
+
+__all__ = [
+    'Action',
+    'IntegerIdGenerator',
+    'Plan',
+    'PlannedAction',
+]
+
+
+class IntegerIdGenerator(object):
```

Comment:
Would it make sense to make this thread-safe, like the `SequenceCounter` in rrc?

Reply:
yep

Reply:
Is this new version better?